plugins/jdbc-adapter-framework/src/main/java/org/polypheny/db/adapter/jdbc/stores/AbstractJdbcStore.java

```diff
@@ -119,254 +138,270 @@ public void createUdfs() {
-                    .append( dialect.quoteIdentifier( partitionPlacement.physicalSchemaName ) )
-                    .append( "." )
-                    .append( dialect.quoteIdentifier( partitionPlacement.physicalTableName ) );
-            builder.append( " DROP " ).append( dialect.quoteIdentifier( columnPlacement.physicalColumnName ) );
-            executeUpdate( builder, context );
-        }
+    public void dropColumn( Context context, long allocId, long columnId ) {
+        PhysicalTable table = adapterCatalog.fromAllocation( allocId );
+        PhysicalColumn column = adapterCatalog.getColumn( columnId, allocId );
+        StringBuilder builder = new StringBuilder();
+        builder.append( "ALTER TABLE " )
+                .append( dialect.quoteIdentifier( table.namespaceName ) )
+                .append( "." )
+                .append( dialect.quoteIdentifier( table.name ) );
+        builder.append( " DROP " ).append( dialect.quoteIdentifier( column.name ) );
+        executeUpdate( builder, context );
+        adapterCatalog.dropColumn( allocId, columnId );
+
+        updateNativePhysical( allocId );
+    }
+
+
+    protected void updateNativePhysical( long allocId ) {
+        PhysicalTable table = adapterCatalog.fromAllocation( allocId );
+        adapterCatalog.replacePhysical( this.currentJdbcSchema.createJdbcTable( table ) );
     }
 
 
     @Override
-    public void truncate( Context context, CatalogTable catalogTable ) {
+    public void truncate( Context context, long allocId ) {
         // We get the physical schema / table name by checking existing column placements of the same logical table placed on this store.
         // This works because there is only one physical table for each logical table on JDBC stores. The reason for choosing this
         // approach rather than using the default physical schema / table names is that this approach allows truncating linked tables.
-        for ( CatalogPartitionPlacement partitionPlacement : catalog.getPartitionPlacementsByTableOnAdapter( getAdapterId(), catalogTable.id ) ) {
-            String physicalTableName = partitionPlacement.physicalTableName;
-            String physicalSchemaName = partitionPlacement.physicalSchemaName;
-            StringBuilder builder = new StringBuilder();
-            builder.append( "TRUNCATE TABLE " )
-                    .append( dialect.quoteIdentifier( physicalSchemaName ) )
-                    .append( "." )
-                    .append( dialect.quoteIdentifier( physicalTableName ) );
-            executeUpdate( builder, context );
-        }
+        PhysicalTable physical = adapterCatalog.fromAllocation( allocId );
+
+        StringBuilder builder = new StringBuilder();
+        builder.append( "TRUNCATE TABLE " )
+                .append( dialect.quoteIdentifier( physical.namespaceName ) )
+                .append( "." )
+                .append( dialect.quoteIdentifier( physical.name ) );
```

Comment:
I think it would be nicer if the physical name is a List or array with an arbitrary number of components. This would also allow representing path etc. Namespace is a Polypheny term. However, since this is common to all stores, there is no common term that fits all. The problem with a list, however, might be the performance...

Reply:
Yes, this is rather easy to use like this.

Reply:
So, we keep it like that for now. But we should change this in the mext iteration.